src/dune_pkg/rev_store.ml

```diff
@@ -5,39 +5,43 @@ module Display = Dune_engine.Display
 module Re = Dune_re
 open Fiber.O
 
-type t = { dir : Path.t }
+type t =
+  { dir : Path.t
+  ; lock : Fiber.Mutex.t
```

Comment:
What about using a file lock instead? We have an `Flock` module in `Dune_util` that's pretty useful.

Reply:
The issue why I went for an in-memory solution is because the cleanup is simple: the due process ends (cleanly or otherwise - the mutex disappears).

With file-based solutions you have to deal with what happens if the lock file is not cleaned up. I took a look at flock and it seems the lock management is in the kernel, so I assume the kernel would know when to release the lock, which sidesteps that issue. Our API for it is blocking, is there a way to wrap a Fiber to await for the lock being released?

Reply:
You can run the locking in a background thread. But tbh, that's probably not necessary. Feel free to block with a timeout waiting on the lock.

Reply:
Implemented it via polling on a nonblocking lock, I think it's not too bad.